### PR TITLE
Make `ignoreProtoXXXFields` comparison option public

### PR DESCRIPTION
**Pull request by tcsc**
Exposes the `ignoreProtoXXFields` comparison option for use in custom
comparison functions.

### CODE REVIEW

File: lib/services/compare.go [hunk 1]
```diff
@@ -59,9 +59,9 @@ func CompareResources[T any](resA, resB T) int {
 	return Different
 }
 
-// ignoreProtoXXXFields is a cmp.Option that ignores XXX_* fields from proto
+// IgnoreProtoXXXFields is a cmp.Option that ignores XXX_* fields from proto
 // messages.
-func ignoreProtoXXXFields() cmp.Option {
+func IgnoreProtoXXXFields() cmp.Option {
```
Comment (marcoandredinis) [inline lib/services/compare.go:64]: Should we move this to another package?

**Review by marcoandredinis — APPROVED** [review]

File: lib/services/compare.go (hunk 1 above)
Comment (tcsc) [reply lib/services/compare.go:64]: I'm kinda tempted to pull out the whole comparison/reconciler business out of `services`, but I _so_ do not have the time right now. Not sure where it should go, in any case.

**Review by r0mant — APPROVED** [review]

**Review by rosstimothy — COMMENTED** [review]
I don't know that exposing this is a good idea. cmp should not be used in production code, and making this public is only going to make it easier to do so.

**Comment by espadolini** [comment]
Could we find a way to use something other than go-cmp, which documents itself like this?
> It is intended to only be used in tests, as performance is not a goal and it may panic if it cannot compare the values. Its propensity towards panicking means that its unsuitable for production environments where a spurious panic may be fatal.

**Review by r0mant — CHANGES_REQUESTED** [review]
@tcsc Agree with Tim and Edoardo actually. Let's not reinforce anti-patterns, I'm not sure where you were planning to use this method but you could just implement `IsEqual` interface and use [CompareResources](https://github.com/gravitational/teleport/blob/12700c37d4e6e66c279a09082ff8e5d9758e4763/lib/services/compare.go#L37).

Or just use `CompareResources` which already falls back to using `cmd.Equal` (for better or worse) and just not have to export anythying.

**Pull request closed without merging**